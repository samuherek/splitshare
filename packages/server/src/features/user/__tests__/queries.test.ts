import gql from 'graphql-tag';
import createTestServer from '../../../test-utils/server';

const ME = gql`
  query Me {
    me {
      id
      email
      displayName
      photoUrl
    }
  }
`;

describe('user queries', () => {
  test('me: query should return a user with appropriate fields', async () => {
    const { query } = await createTestServer({
      user: { id: 1 },
      models: {
        User: {
          getById: jest.fn(() => ({
            id: 1,
            email: 'email@email.com',
            displayName: null,
            photoUrl: null,
          })),
        },
      },
    });

    const res = await query({ query: ME });

    expect(res).toMatchSnapshot();
  });
});
