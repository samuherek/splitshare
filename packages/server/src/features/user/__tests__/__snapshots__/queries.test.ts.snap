// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user queries me: query should return a user with appropriate fields 1`] = `
Object {
  "data": Object {
    "me": Object {
      "displayName": null,
      "email": "email@email.com",
      "id": "1",
      "photoUrl": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
